mapeditor/inspector/inspector.cpp

```diff
@@ -376,19 +376,22 @@ void Inspector::updateProperties(CGArtifact * o)
 
 	addProperty(QObject::tr("Message"), o->message, false);
 
-	const CArtifactInstance * instance = o->getArtifactInstance();
-	if(instance && o->ID == Obj::SPELL_SCROLL)
+	if(o->ID == Obj::SPELL_SCROLL)
 	{
-		SpellID spellId = instance->getScrollSpellID();
-		if(spellId != SpellID::NONE)
+		const CArtifactInstance * instance = o->getArtifactInstance();
+		if (instance)
 		{
-			auto * delegate = new InspectorDelegate;
-			for(auto const & spell : LIBRARY->spellh->objects)
+			SpellID spellId = instance->getScrollSpellID();
+			if(spellId != SpellID::NONE)
 			{
-				if(controller.map()->allowedSpells.count(spell->id) != 0)
-					delegate->options.push_back({QObject::tr(spell->getNameTranslated().c_str()), QVariant::fromValue(int(spell->getId()))});
+				auto * delegate = new InspectorDelegate;
+				for(auto const & spell : LIBRARY->spellh->objects)
+				{
+					if(controller.map()->allowedSpells.count(spell->id) != 0)
+						delegate->options.push_back({QObject::tr(spell->getNameTranslated().c_str()), QVariant::fromValue(int(spell->getId()))});
+				}
+				addProperty(QObject::tr("Spell"), LIBRARY->spellh->getById(spellId)->getNameTranslated(), delegate, false);
 			}
```

Comment:
It would be nice if this didn't indent as much. For example:
```c++
if(o->ID != Obj::SPELL_SCROLL) return;

const CArtifactInstance * instance = o->getArtifactInstance();
if (!instance) return;

SpellID spellId = instance->getScrollSpellID();
if(spellId == SpellID::NONE) return;

auto * delegate = new InspectorDelegate;
for(auto const & spell : LIBRARY->spellh->objects)
{
	if(controller.map()->allowedSpells.count(spell->id) != 0)
		delegate->options.push_back({QObject::tr(spell->getNameTranslated().c_str()), QVariant::fromValue(int(spell->getId()))});
}
addProperty(QObject::tr("Spell"), LIBRARY->spellh->getById(spellId)->getNameTranslated(), delegate, false);
```